provider_shopper/lib/main.dart

```diff
@@ -27,9 +27,11 @@ class MyApp extends StatelessWidget {
         // of ChangeNotifierProvider. Moreover, CartModel depends
         // on CatalogModel, so a ProxyProvider is needed.
         ChangeNotifierProxyProvider<CatalogModel, CartModel>(
-          create: (context) => CartModel.empty(),
-          update: (context, catalog, previousCart) =>
-              CartModel(catalog, previousCart),
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            cart.catalog = catalog;
```

Comment:
Nit but I'd use the `..` operator instead of making a block

Reply:
To offer a counterpoint, in a sample that's aimed at beginners (like this one), we sometimes stretch out the syntax to maximize readability for new folks.

Entirely your call which way to go, though, Filip.

Reply:
Yes, I had the cascade operator there at first but went with the more verbose syntax for the reason @RedBrogdon mentions. We're not teaching Dart niceties here, so I go with as "vanilla" syntax as possible.

Reply:
@filiph @RedBrogdon Thanks a ton, As a beginner user it's very hard to understand these kind of syntax `..` 
vanilla syntax rocks:)